extensions/kafka/deployment/src/main/java/org/apache/camel/quarkus/component/kafka/deployment/KafkaProcessor.java

```diff
@@ -68,4 +69,12 @@ public void configureKafkaComponentForDevServices(
             }
         }
     }
+
+    @BuildStep
+    public ReflectiveClassBuildItem reflectiveClasses() {
+        return new ReflectiveClassBuildItem(false, true,
+                "org.apache.kafka.clients.producer.KafkaProducer",
+                "org.apache.kafka.clients.consumer.KafkaConsumer",
+                "org.apache.kafka.clients.producer.internals.Sender");
+    }
```

Comment:
There are [those lines](https://github.com/apache/camel/blob/main/components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaProducer.java#L170..L171)  where kafkaProducer is of type `org.apache.kafka.clients.producer.Producer kafkaProducer;`. Is it included in test coverage ? Maybe it needs native registration also ?

Reply:
Maybe most of the time the underlying type is `org.apache.kafka.clients.producer.KafkaProducer`. However, could we have another implementation used when using the [setter](https://github.com/apache/camel/blob/main/components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaProducer.java#L138) ?

Reply:
Good point! I'm not very sure why we need to use reflection here since ```org.apache.kafka.clients.producer.Producer``` is an interface which does not have ```clientId``` field. So it probably can remove the reflection usage because ```clientId``` is alway from the property ```client.id```. Is it right ?

Reply:
About knowing whether registering the class for reflection is useful. The best I can see is to have it covered in a native test.

About knowing whether using reflection is really needed in camel-kafka. `kafkaProducer.getClass()` might return `org.apache.kafka.clients.producer.KafkaProducer` most of the time which has field `clientId`.

@davsclaus: Could we safely assume that `kafkaProducer.getClass()` always returns `org.apache.kafka.clients.producer.KafkaProducer` [here](https://github.com/apache/camel/blob/main/components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaProducer.java#L170..L171) ? What happen if someone change the implementation by using the [setter](https://github.com/apache/camel/blob/main/components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaProducer.java#L138) ?

Reply:
yeah, I will check it and I think the current native test should cover it. But it igores the Exception, it might be useful to show a warning message.

Reply:
Hi @aldettinger , I change to add all implementations of ```org.apache.kafka.clients.producer.Producer``` and ```org.apache.kafka.clients.consumer.Consumer``` to register as RefelctiveClass. Also add a body check for ```client.id```. It should work even if we change an implementation in the future. Anyway it must have the field ```clientId```.

Reply:
OK, all issues from this conversation must be addressed now. The failing issues from CI seems are not related. Reading the conversations again, the next step might be to figure out why shutdownTimeout is not working in this case. Keep up the good work :+1: